### PR TITLE
feat: detect email collisions

### PR DESCRIPTION
**Pull request by yeagerd**
1. Webhook Schema Improvements (services/user/schemas/webhook.py)
* Type annotation enhancement: Added proper type hints (Any, Dict, List) for better type safety
* Email handling refactor:
  * Changed email_addresses field from Optional[str] to Optional[List[Dict[str, Any]]] to properly handle Clerk's email array structure
  * Removed the @field_validator decorator and moved logic to the primary_email property
  * Improved email extraction logic with better null checking

2. Enhanced Webhook Service Logic (services/user/services/webhook_service.py)
* Email update functionality: Added logic to update user emails when they change in Clerk
* Email collision detection: Added checks to prevent email conflicts between different external auth IDs
* Improved idempotency: Enhanced existing user detection and handling with more detailed response actions
* Better error handling: More specific error messages and logging for different scenarios

3. Comprehensive Test Coverage (services/user/tests/test_webhook_service.py)
* Updated test assertions: Changed expected actions from "user_creation_skipped" to "user_already_exists"
* New test case: Added test_update_external_auth_id_when_email_exists() to test email collision scenarios
* Enhanced test coverage: More thorough testing of webhook edge cases and user update scenarios

### CODE REVIEW

File: services/user/utils/email_collision.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+            # Check for collision
+            collision_user = await self.check_collision(normalized_email)
+
+            if collision_user:
+                return {
+                    "available": False,
+                    "normalized_email": normalized_email,
+                    "collision": True,
+                    "existing_user_id": collision_user.external_auth_id,
```
Comment (yeagerd) [inline services/user/utils/email_collision.py:125]: TODO: see how to mitigate security risk
